libs/ledger/src/chain/main_chain.cpp

```diff
@@ -1477,6 +1478,17 @@ BlockStatus MainChain::InsertBlock(IntBlockPtr const &block, bool evaluate_loose
     }
   }
 
+  if (evaluate_loose_blocks)  // normal case - not being called from inside CompleteLooseBlocks
```

Comment:
this is a performance improvement that avoids detecting duplicate TXs unneccessarily